Entities/Natural/Trees/AppleTreeLogic.as

```diff
@@ -0,0 +1,384 @@
+// Apple tree Logic
+
+#include "TreeSync.as"
+
+void onInit(CBlob@ this)
+{
+	InitVars(this);
+	this.server_setTeamNum(-1);
+	TreeVars vars;
+	vars.seed = XORRandom(30000);
+	vars.growth_time = 250 + XORRandom(30);
+	vars.height = 0;
+	vars.max_height = 2 + XORRandom(2);
+	vars.grown_times = 0;
+	vars.max_grow_times = 50;
+	vars.last_grew_time = getGameTime() - 1; //pretend we started a frame ago ;)
+	InitTree(this, vars);
+	this.set("TreeVars", vars);
+	
+	// Produce
+	if (this.hasTag("instant_grow"))
+	{
+		GrowApples(this);
+	}
+}
+
+// Grow the apples
+void GrowApples(CBlob@ this)
+{
+	if (getNet().isServer())
+	{
+		for (int i = 0; i < 2; i++)
+		{	
+			switch (i)
+			{
+				// Apple 1
+				case 0: InitFirstApple(this); break;
+	
+				// Apple 2
+				case 1: InitSecondApple(this); break;
+			}
+		}
+	}
+}
+
+// Workaround to odd switch requirements ("cannot set a variable in a switch statement") logic in void GrowApples
+// Apple 1
+void InitFirstApple(CBlob@ this)
+{
+	// pos
+	Vec2f offset = this.getPosition();
+	int v = this.isFacingLeft() ? 0 : 1;
+	offset = (offset + Vec2f(8 + v, -35));
+	
+	// create
+	CBlob@ apple1 = server_CreateBlob('apple', -1, offset);
+	if (apple1 !is null)
+	{
+		// make it referable
+		this.set_u16("apple1", apple1.getNetworkID());
+
+		CShape@ shape1 = apple1.getShape();
+		
+		// the apple is on the tree, so make it hang/static
+		if (shape1 !is null)
+		{
+			shape1.SetStatic(true);
+		}
+	}
+}
+
+// Apple 2
+void InitSecondApple(CBlob@ this)
+{
+	// pos
+	Vec2f offset = this.getPosition();
+	int v = this.isFacingLeft() ? 0 : 1;
+	offset = (offset + Vec2f(-8 + v, -33));

```

Comment:
This line is the only difference between the init apple functions? That is redundant. Remove one and add a second parameter such as `uint8 apple_id` -> `if (apple_id == 0) { // set offset for the first apple here` `else if (apple_id == 1) { // set offset for the secoind apple here` if you do this you *could* use the for loop on line 32 (i.e. `InitApple(this, i)`) but the switch statement still serves no purpose.